filebeat/prospector/prospector.go

```diff
@@ -77,7 +82,7 @@ func (p *Prospector) initProspectorer(outlet channel.Outleter, states []file.Sta
 	case harvester.RedisType:
 		prospectorer, err = redis.NewProspector(config, outlet)
 	case harvester.LogType:
-		prospectorer, err = log.NewProspector(config, states, outlet, p.done)
+		prospectorer, err = log.NewProspector(config, states, outlet, p.done, p.beatDone)
```

Comment:
well, the matter is a little more complicated. We do have 3 channels for signaling shutdown. In the past we used to have 4 or more signalers (I got rid of spooler and publisher) :)

The outDone channel (filebeat.go) replaces the publisher closing, and is therefore closed much later. The beatDone channel is require when creating the stateOutlet, as this one needs to be unblocked when `fb.done` is closed (otherwise the crawler will not stop).